code/datums/components/durability.dm

```diff
@@ -0,0 +1,189 @@
+
+	/// If present, a sound that plays when the item breaks.
+	var/break_sound
+
+	/// The force of the item after breaking.
+	var/broken_force
+
+	/// A flat penalty to apply to the force of the item after breaking.
+	var/broken_force_decrease
+
+	/// The throw force of the item after breaking.
+	var/broken_throw_force = 0
+
+	/// A flat penalty to apply to the throw force of the item after breaking.
+	var/broken_throw_force_decrease = 0
+
+	/// The throw range of the item after breaking.
+	var/broken_throw_range
+
+	/// The embedding attributes of the item after breaking.
+	var/list/broken_embedding
+
+	/// The block chance of the item after breaking.
+	var/broken_block_chance
+
+	/// The `w_class` of the item after breaking.
+	var/broken_w_class
+
+	/// If present, a message that plays when the item breaks.
+	var/broken_message
+
+/datum/component/durability/Initialize(
+	broken,
+	broken_prefix,
+	broken_name,
+	broken_desc,
+	broken_embedding,
+	max_durability,
+	worn_durability,
+	current_durability,
+	worn_icon_state,
+	broken_icon_state,
+	break_sound,
+	broken_force,
+	broken_force_decrease,
+	broken_throw_force,
+	broken_throw_force_decrease,
+	broken_throw_range,
+	list,
+	broken_block_chance,
+	broken_message,
+	broken_w_class
+)
+	if (!isnull(broken))
+		src.broken = broken
+
+	if (!isnull(broken_prefix))
+		src.broken_prefix = broken_prefix
+
+	if (!isnull(broken_name))
+		src.broken_name = broken_name
+
+	if (!isnull(broken_desc))
+		src.broken_desc = broken_desc
+
+	if (!isnull(max_durability))
+		src.max_durability = max_durability
+		src.current_durability = max_durability
+
+	if (!isnull(current_durability))
+		src.current_durability = current_durability
+
+	if (!isnull(broken_icon_state))
+		src.broken_icon_state = broken_icon_state
+
+	if (!isnull(break_sound))
+		src.break_sound = break_sound
+
+	if (!isnull(broken_force))
+		src.broken_force = broken_force
+
+	if (!isnull(broken_throw_force))
+		src.broken_throw_force = broken_throw_force
+
+	if (!isnull(broken_throw_force_decrease))
+		src.broken_throw_force_decrease = broken_throw_force_decrease
+
+	if (!isnull(broken_throw_range))
+		src.broken_throw_range = broken_throw_range
+
+	if (!isnull(broken_embedding))
+		src.broken_embedding = broken_embedding
+
+	if (!isnull(broken_block_chance))
+		src.broken_block_chance = broken_block_chance
+
+	if (!isnull(broken_message))
+		src.broken_message = broken_message
+
+	if (!isnull(broken_w_class))
+		src.broken_w_class = broken_w_class
+
+
+/datum/component/durability/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_ITEM_AFTERATTACK, PROC_REF(on_after_attack))
+
+/datum/component/durability/UnregisterFromParent()
+	UnregisterSignal(parent, list(COMSIG_ITEM_AFTERATTACK))
+
+/datum/component/durability/proc/on_after_attack(obj/item/source, atom/target, mob/living/user, proximity)
+	SIGNAL_HANDLER
+
+	if(broken)
+		return NONE
+
+	current_durability -= 1
+
+	if(!isnull(max_durability) && current_durability == 0)
```

Comment:
```suggestion
	if(current_durability > 0)
		return
```
Two things:
- Max durability should never be null, in that case this component is pointless. Add a `CRASH` case in Initialize if that is true, or just give it a default value.
- We should return early instead of having the entire rest of the proc be indented.